app/controllers/comments/create.js

```diff
@@ -15,15 +15,26 @@ export const createComment = async (data) => {
     }
   }
 
-  const created = await db.Comments.create({
-    data: {
-      Questions: {
-        connect: {
-          question_id: value.questionId,
-        },
+
+  const commentData = {
+    Questions: {
+      connect: {
+        question_id: value.questionId,
       },
-      comment: value.comment,
+    },
+    comment: value.comment,
+  };
+
+  if (value.user.userEmail) {
```

Comment:
Got it, I switched the conditional to use that isAnonymous value as indicated, I also added a prior validation:

`if (!value.isAnonymous && !value.user.userEmail) {
    return {
      error: {
        message: INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE,
        detail: "The comment is not anonymous but no user email was provided in the user object",
      }
    }
  }` 

This is necessary because the 'userEmail' is not enforced as required inside the user object, so we could not provide 'isAnonymous' or provide as false, and also omit 'userEmail', but this check fixes that, thank you!